torch/distributed/tensor/_random.py

```diff
@@ -198,7 +198,7 @@ def set_offset(self, name: str, offset: int) -> None:
             )
 
         seed_tensor = (self.rng_states[name])[0:8]
-        offset_tensor = torch.tensor([offset]).view(torch.uint8)
+        offset_tensor = torch.tensor([offset], dtype=torch.uint64).view(torch.uint8)
```

Comment:
Isn't this immediately cast to uint8?

Reply:
offset comes in as a uint64 essentially.  we're using a tensor to convert it to its constituent bytes because this is the format used by torch's RNG state.  the initial dtype=uint64 is needed to prevent overflow warnings mainly.